contracts/aggregator/mainnet/flashloan/helpers.sol

```diff
@@ -119,27 +135,31 @@ contract Helper is Variables {
     }
 
     /**
-     * @dev Supply token for the amount to compound.
-     * @notice Supply token for the amount to compound.
-     * @param _token token addresses.
-     * @param _amount amount of token.
+     * @dev Supply tokens for the amounts to compound.
+     * @notice Supply tokens for the amounts to compound.
+     * @param _tokens token addresses.
+     * @param _amounts amounts of tokens.
     */
-    function compoundSupply(address _token, uint256 _amount) internal {
-        address[] memory cTokens_ = new address[](1);
-        if (_token == wEthToken) {
-            wEth.withdraw(_amount);
-            CEthInterface cEth_ = CEthInterface(cEthToken);
-            cEth_.mint{value: _amount}();
-            cTokens_[0] = cEthToken;
-        } else {
-            CTokenInterface cToken_ = CTokenInterface(tokenToCToken[_token]);
-            // Approved already in addTokenToctoken function
-            require(cToken_.mint(_amount) == 0, "mint failed");
-            cTokens_[0] = tokenToCToken[_token];
-        }
-        uint[] memory errors_ = troller.enterMarkets(cTokens_);
-        for(uint i=0; i < errors_.length; i++){
-            require(errors_[i] == 0, "Comptroller.enterMarkets failed.");
+    function compoundSupply(address[] memory _tokens, uint256[] memory _amounts) internal {
+        uint256 length_ = _tokens.length;
+        require(_amounts.length == length_, "array-lengths-not-same");
+        address[] memory cTokens_ = new address[](length_);
+        for(uint256 i = 0; i < length_; i++) {
+            if (_tokens[i] == wEthToken) {
+                wEth.withdraw(_amounts[i]);
+                CEthInterface cEth_ = CEthInterface(cEthToken);
+                cEth_.mint{value: _amounts[i]}();
+                cTokens_[i] = cEthToken;
+            } else {
+                CTokenInterface cToken_ = CTokenInterface(tokenToCToken[_tokens[i]]);
+                // Approved already in addTokenToctoken function
+                require(cToken_.mint(_amounts[i]) == 0, "mint failed");
+                cTokens_[i] = tokenToCToken[_tokens[i]];
+            }
+            uint256[] memory errors_ = troller.enterMarkets(cTokens_);
```

Comment:
Why are we entering the market on every call?

Reply:
Removed